### PR TITLE
Preserve velocities in HREX permutation moves

### PR DESCRIPTION
**Pull request by mcwitt**
Currently velocities are lost after each MD propagation step in HREX, with the subsequent MD propagation step starting with zero velocity. This is effectively freezing the system every 5 ps with the current parameters.

This PR switches from using `CoordsBox` as the replica state type to `CoordsVelBox` to avoid dropping the velocities during HREX permutation moves.

### CODE REVIEW

File: timemachine/fe/free_energy.py [hunk 1]
```diff
@@ -374,15 +378,16 @@ def sample(
 
     assert np.all(np.isfinite(ctxt.get_x_t())), "Equilibration resulted in a nan"
 
-    def run_production_steps(n_steps: int) -> Tuple[NDArray, NDArray]:
+    def run_production_steps(n_steps: int) -> Tuple[NDArray, NDArray, NDArray]:
         coords, boxes = ctxt.multiple_steps(
             n_steps=n_steps,
             store_x_interval=md_params.steps_per_frame,
         )
+        final_velocities = ctxt.get_v_t()
```
Comment (proteneer) [inline timemachine/fe/free_energy.py:386]: sanity check, does `ctxt.get_x_t()` match `coords[-1]`? i.e. making sure that the last frame of coords are in sync with the velocities.

File: timemachine/fe/free_energy.py [hunk 2]
```diff
@@ -404,7 +409,10 @@ def run_production_local_steps(n_steps: int) -> Tuple[NDArray, NDArray]:
             )
             coords.append(x_t)
             boxes.append(box_t)
-        return np.concatenate(coords), np.concatenate(boxes)
+
+        final_velocities = ctxt.get_v_t()
+
```
Comment (proteneer) [inline timemachine/fe/free_energy.py:414]: same sanity check, does ctxt.get_x_t() match coords[-1]? i.e. making sure that the last frame of coords are in sync with the velocities.

File: timemachine/fe/free_energy.py [hunk 3]
```diff
@@ -351,7 +353,9 @@ def sample(
 
     Returns
     -------
-    xs, boxes: np.arrays with .shape[0] = md_params.n_frames
+    xs, boxes: NDArray with .shape[0] = md_params.n_frames
+
+    final_velocities: NDArray with shape (N,)
```
Comment (proteneer) [inline timemachine/fe/free_energy.py:358]: This docstring is technically incorrect, I think this function returns a single tuple with 3 components, (xs, boxes, final_velocities), as opposed to 2 components ((xs, boxes), final_velocities).

Comment (mcwitt) [reply timemachine/fe/free_energy.py:358]: Ah yeah, the grouping of `xs, boxes` wasn't meant to indicate they were in a separate tuple. Will split lines to clarify.

Comment (mcwitt) [reply timemachine/fe/free_energy.py:360]: Fixed in https://github.com/proteneer/timemachine/pull/1142/commits/2fe7c2fb85aac0405b8cae193280760ed8e209a8

File: timemachine/fe/free_energy.py (hunk 1 above)
Comment (mcwitt) [reply timemachine/fe/free_energy.py:389]: Assertion added in [d151e39](https://github.com/proteneer/timemachine/pull/1142/commits/d151e39490322bf88df7f07989cad682fa6fa785)

File: timemachine/fe/free_energy.py (hunk 2 above)
Comment (mcwitt) [reply timemachine/fe/free_energy.py:418]: Assertion added in [d151e39](https://github.com/proteneer/timemachine/pull/1142/commits/d151e39490322bf88df7f07989cad682fa6fa785)

**Review by jkausrelay — APPROVED** [review]

**Review by proteneer — APPROVED** [review]

**Pull request merged**